File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if(head == nullptr){
            return head;
        }
        if(head->next == nullptr){
            return head;
        }
        int currVal = head->val;
        ListNode* start = head;
        ListNode* holder = head;
        start = start->next;
        while(start!=nullptr){
            if(start->val == currVal){
                start = start->next;
            }else{
                holder->next = start;
                holder = start;
                currVal = start->val;
                start = start->next;
            }
        }
        holder ->next = nullptr;
        return head;
    }
};

File: 0083-remove-duplicates-from-sorted-list/0083-remove-duplicates-from-sorted-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if(head==null || head.next==null){
            return head;
        }else{
            ListNode holder = head;
            ListNode first = head;
            int temp = first.val;
            while(head.next!=null){
                head = head.next;
                if(temp == head.val){
                    if(head.next==null){
                        first.next = null;
                    }
                }else{
                    first.next = head;

                    first = head;
                    temp = head.val;
                }
            }
            head = holder;
            return head;
        }
    }
}